transport/owr_crypto_utils.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2015, Ericsson AB. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or other
+ * materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ */
+
+/*/
+\*\ Owr crypto utils
+/*/
+
+/**
+ * 
+ *
+ * Functions to get certificate, fingerprint and private key.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "owr_crypto_utils.h"
+#include "owr_private.h"
+#include "owr_utils.h"
+
+
+#ifdef OWR_STATIC
+#include <stdlib.h>
+#endif
+
+#ifdef __APPLE__
+#include <TargetConditionals.h>
+#endif
+
+#ifdef __ANDROID__
+#include <android/log.h>
+#endif
+
+#include <openssl/ssl.h>
+#include <openssl/evp.h>
+#include <openssl/rsa.h>
+
+/* PUBLIC */
+
+/**
+ * owr_crypto_create_crypto_data:
+ * @types:
+ * @callback: (scope async):
+ */
+
+/**
+ * OwrCaptureSourcesCallback:
+ * @privatekey: (transfer none)
+ * @certificate: (transfer none)
+ * @user_data: (allow-none): 
+ *
+ * Prototype for the callback passed to owr_get_capture_sources()
+ */
+
+
+void owr_crypto_create_crypto_data(OwrCryptoDataCallback callback)
+{
+
```

Comment:
The style of this file is not in line with the rest of the project. But I can fix that quickly post-merge.